### PR TITLE
Update Coverage for Ruby 3.2 interface changes.

### PR DESCRIPTION
**Pull request by mtortonesi**
This patch request should fix issue #3149.

Thanks to @ioquatix and @eregon.

### CODE REVIEW

File: lib/truffle/coverage.rb [hunk 1]
```diff
@@ -10,13 +10,23 @@
...
 
-  def self.start
+  def self.supported?(mode)
+    mode == :lines
+  end
+
+  def self.start(*arguments, **options)
+    # We have to track if the :lines option was provided, as that calls for a
+    # different result format
+    @lines = true if options[:lines]
```
Comment (ioquatix) [inline lib/truffle/coverage.rb:20]:
```suggestion
@lines = !!options[:lines]

**Comment by ioquatix** [comment]
For `covered`, we also need to support:

```ruby
::Coverage.result(stop: false, clear: true)
```

**Comment by mtortonesi** [comment]
Dear @ioquatix, I changed the code as per your request. 

Would be happy to help you with adding support for ::Coverage.result(stop: false, clear: true) but I need more information about it.

**Comment by ioquatix** [comment]
```ruby
   def self.result(stop: true, clear: true)
     result = peek_result
     Truffle::Coverage.disable if stop || clear
     Truffle::Coverage.enable unless stop
```

I believe something like this can work.. but it's just a guess.

**Comment by eregon** [comment]
Thank you for this this.
What would be super helpful is if you could write specs under `spec/ruby/library/coverage` for all the edge cases we found in https://github.com/oracle/truffleruby/pull/3150.

**Comment by eregon** [comment]
Having these specs is needed to ensure we implement working and compatible behavior + avoid regressions.
For instance we need specs for arguments and keywords arguments given to `Coverage.start`.

**Comment by andrykonchin** [comment]
I will add missing specs and prepare the PR to merging.

**Comment by eregon** [comment]
@mtortonesi Sorry it takes a while but as you can see it's a lot more complicated than the initial patch.
Ah and because this was a PR from your master, it might have been unintentionally closed as a side effect.

We'll merge the changes when they are ready, we have an internal PR with extra specs etc.

**Comment by eregon** [comment]
Fixed in https://github.com/oracle/truffleruby/pull/3389

**Pull request closed without merging**